### DISCUSSION TITLE
unable to run https server over linode

### DISCUSSION THREAD

**Discussion by nikhil-varma-mudunuri** [pocketbase/pocketbase#4027] [Q&A] [answered]
when i ran
./pocketbase serve --https="addr:443"

i am getting this error when accessing it 
http: TLS handshake error from addr:31577: acme/autocert: missing server name

**Comment by ganigeorgiev** [comment, marked as answer]
Let's encrypt doesn't support issuing certificates on bare IPs, aka. you need a valid public domain name if you want to use the autogenerated Let's encrypt certificate.

Once you have a domain name and link it to your VPS, you can just run `./pocketbase serve yourdomain.com`

**Comment by nikhil-varma-mudunuri** [comment]
when i use this ./pocketbase serve --https="nanme.in:445"

i am getting below error
 http: TLS handshake error from xxx.xx.xx.xx:62144: acme/autocert: unable to satisfy "https://acme-v02.api.letsencrypt.org/acme/authz-v3/296979375186" for domain "nanme.in": no viable challenge type found

**Reply by ganigeorgiev** [reply]
I'm starting to regret introducing too many flags as it seems that users don't understand what they are doing and I'm failing to explain it.

Couple things:

1. The Let's encrypt certificate validation needs to be performed on either 80 or 443 port (or a special DNS record for the [dns-01 challenge](https://letsencrypt.org/docs/challenge-types/#dns-01-challenge))

2. Let's encrypt recommends to have both 80 and 443 open - https://letsencrypt.org/docs/allow-port-80/.

3. If PocketBase is the only application running on your server, prefer using just `./pocketbase serve nanme.in`.
   This command is equivalent to:

   ```sh
   # 0.0.0.0 is a special address that means bind to all available IPv4 interfaces
   ./pocketbase serve nanme.in --http="0.0.0.0:80" --https="0.0.0.0:443"
   ```

   When you use only the `--https` option this binds **only** to the IP that your domain name is resolving, which sometimes may require updating your `/etc/hosts` file (for more details see https://github.com/pocketbase/pocketbase/discussions/3190).

If you want to host multiple applications that binds to 80 and 443 ports on the same server, then my recommendation is to use a reverse proxy like nginx, apache, caddy - https://pocketbase.io/docs/going-to-production/#using-reverse-proxy.

**Comment by nikhil-varma-mudunuri** [comment]
@ganigeorgiev 
here is the context for #316 
comand i used: ./pocketbase serve name.in

running this command in linode i have created in debian based system
name.in domain names resides in godaddy  so i have changed nameservers to linode nameservers

able to run http version but not https version

**Reply by ganigeorgiev** [reply]
It is not clear what actually is not working, what you are running, why did you try to bind to 445 and not 443, etc.
Your reported initial error doesn't correspond to the commands you are executing. Please be more specific.
It is really frustrating reading this type of questions and I'm getting really tired by this.
I tried to simplify with https://github.com/pocketbase/pocketbase/discussions/3190 but it seems that it didn't help and I'm out of ideas how to make it more simpler than this.

- Is `name.in` your actual domain you are testing with?
Are you sure that your nameservers change and the dns A records have actually propagated globally?
You can verify the latter by checking for example in https://www.whatsmydns.net/#A/name.in.

- Does your linode server comes with IPv4 or IPv6 address? If you have only IPv6 address then you may need to specify the IPv6 equivalent of `0.0.0.0`:
  ```js
  ./pocketbase serve example.com --http="[::]:80" --https="[::]:443"
  ```

**Comment by nikhil-varma-mudunuri** [comment]
okay, leave

**Reply by nikhil-varma-mudunuri** [reply]
just want to clarify for you,
my actual domain is `nikhilvarma.in`
i have checked in this website https://www.whatsmydns.net
both A and nameservers  are propagated as i expected

i don't know why it's still not working

**Reply by ganigeorgiev** [reply]
1. Neither the A, nor the NS records have propagated everywhere.

2. You are ignoring my previous questions. If it is not working, please report what command you are running, what is the error you are getting, what do you have installed on the server, etc. Try to provide the exact steps you've taken to reach to the error. By just saying "it's still not working" without any details what actually is not working I don't know how to help.

**Comment by nikhil-varma-mudunuri** [comment]


**Reply by ganigeorgiev** [reply]
Are you sure that it is not some browser cache you are seeing?

Is there any terminal (stdout/stderr) logs printed after starting the PocketBase executable and trying to access the address?

Edit: I see that you added the TLS handshake error but this only indicates that you or someone else is trying to access your website via the IP and not the domain name.

**Reply by ganigeorgiev** [reply]
I've also just opened https://nikhilvarma.in/ and it is loading for me so it is definetelly either a cache or some other local configuration.

**Reply by nikhil-varma-mudunuri** [reply]
yeah i also tried in new private tab it worked , thankssss
